log_extractor/files.py

```diff
@@ -47,7 +47,7 @@ def open(self, filepath):
 
             if filepath.endswith(".gz"):
                 return gzip.open(
-                    extracted_filepath, "r:gz"
+                    filename=extracted_filepath, mode="rb"
```

Comment:
Do you change it because of PY3? Will it work for PY2 as well?

Reply:
Yes, it fail in python3 as followed:

(.art) [@localhost log-extractor]$ log-extractor --source <jenkins_url>/6 --team <team_name> --logs engine.log,vdsm.log
Traceback (most recent call last):
  File "/<...>/.art/bin/log-extractor", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/<....>/.art/lib/python3.7/site-packages/log_extractor/extractor.py", line 578, in run
    log_extractor.parse_logs()
  File "/<....>/.art/lib/python3.7/site-packages/log_extractor/extractor.py", line 440, in parse_logs
    with closing(tar_object.open(log_file)) as f:
  File "<...>/.art/lib/python3.7/site-packages/log_extractor/files.py", line 50, in open
    extracted_filepath, "r:gz"
  File "/usr/lib64/python3.7/gzip.py", line 53, in open
    binary_file = GzipFile(filename, gz_mode, compresslevel)
  File "/usr/lib64/python3.7/gzip.py", line 163, in __init__
    fileobj = self.myfileobj = builtins.open(filename, mode or 'rb')
ValueError: invalid mode: 'r:gzb'

And with this patch, it runs successfully, with PY3 and PY2 as well

Roni